### PR TITLE
Various sanitycheck enhancements/fixes

### PR DESCRIPTION
**Pull request by nashif**
- Add timestamps
- fix reporting: build vs. run
- show handler in verbose mode

### CODE REVIEW

File: scripts/sanitycheck [hunk 1]
```diff
@@ -1224,23 +1225,29 @@ class MakeGenerator:
         handler = None
         if type == "qemu":
             handler = QEMUHandler(instance)
+            handler.type = "qemu"
```
Comment (marc-hb) [inline scripts/sanitycheck]: Make the .type attribute mandatory by adding it to the constructors instead:
```
handler = QEMUHandler(instance, type_str)
```
It could be used for a future `__str__()`

(BTW "type" is a builtin, so not the best name)

File: scripts/sanitycheck [hunk 2]
```diff
@@ -3020,9 +3027,14 @@ def chatty_test_cb(instances, goals, goal):
     else:
         status = goal.make_state
 
-    info("{:>{}}/{} {:<25} {:<50} {}".format(
+    if goal.handler and goal.handler.type:
```
Comment (marc-hb) [inline scripts/sanitycheck]: Once the string is mandatory you can just:
```python
if goal.handler:
      handler_type = goal.handler.type_name
else:
      handler_type = "build"
```
or shorter:
```python
info( ..., goal.handler.type_name if goal.handler else "build")
```

File: scripts/sanitycheck [hunk 3]
```diff
@@ -1037,7 +1037,7 @@ class MakeGoal:
             return self.build_log
         elif self.make_state == "running":
             # Failure in sub-make for "make run", qemu probably failed to start
-            return self.run_log
+            return self.handler_log
```
Comment (marc-hb) [inline scripts/sanitycheck]: Is this related to the other change at line 1365+? Sorry I'm lost.

**Review by marc-hb — COMMENTED** [review]
see inline

File: scripts/sanitycheck [hunk 4]
```diff
@@ -418,9 +419,11 @@ class ExecutionError(MakeError):
 
 log_file = None
 
-
 # Debug Functions
-def info(what):
+def info(what, show_time=True):
```
[the remark this answers is not in the document]
Comment (ceolin) [reply scripts/sanitycheck]: indeed, it may worth re-factoring it some time,

File: scripts/sanitycheck (hunk 1 above)
Comment (ceolin) [reply scripts/sanitycheck]: +1

**Review by ceolin — APPROVED** [review]
Would be nice to have all improvements @marc-hb mentioned. As remaining problems were not introduced in this patch ... +2
